libs/agno/agno/a2a/client.py

```diff
@@ -0,0 +1,599 @@
+"""A2A (Agent-to-Agent) protocol client for Agno.
+
+This module provides a Pythonic client for communicating with any A2A-compatible
+agent server, enabling cross-framework agent communication.
+
+"""
+
+import json
+from typing import Any, AsyncIterator, Dict, List, Optional
+from uuid import uuid4
+
+from agno.a2a.exceptions import (
+    A2AAgentNotFoundError,
+    A2AConnectionError,
+    A2AError,
+    A2ARequestError,
+    A2ATaskFailedError,
+    A2ATimeoutError,
+)
+from agno.a2a.schemas import AgentCard, Artifact, StreamEvent, TaskResult
+from agno.media import File, Image
+from agno.utils.log import log_warning
+
+try:
+    from httpx import AsyncClient, Client, HTTPStatusError, TimeoutException
+except ImportError:
+    raise ImportError("`httpx` not installed. Please install using `pip install httpx`")
+
+
+__all__ = ["A2AClient"]
+
+
+class A2AClient:
+    """Async client for A2A (Agent-to-Agent) protocol communication.
+
+    Provides a Pythonic interface for communicating with any A2A-compatible
+    agent server, including Agno AgentOS with a2a_interface=True.
+
+    The A2A protocol is a standard for agent-to-agent communication that enables
+    interoperability between different AI agent frameworks.
+
+    Attributes:
+        base_url: Base URL of the A2A server
+        timeout: Request timeout in seconds
+        a2a_prefix: URL prefix for A2A endpoints (default: "/a2a")
+
+    """
+
+    def __init__(
+        self,
+        base_url: str,
+        timeout: int = 30,
+        a2a_prefix: str = "/a2a",
+        json_rpc_endpoint: Optional[str] = None,
+    ):
+        """Initialize A2AClient.
+
+        Args:
+            base_url: Base URL of the A2A server (e.g., "http://localhost:7777")
+            timeout: Request timeout in seconds (default: 30)
+            a2a_prefix: URL prefix for A2A endpoints (default: "/a2a")
+            json_rpc_endpoint: Optional single endpoint for all JSON-RPC calls.
+                If set, all A2A methods will POST to this endpoint instead of
+                individual REST endpoints. Use "/" for Google ADK compatibility.
+                Example: json_rpc_endpoint="/" for Google ADK servers.
+        """
+        self.base_url = base_url.rstrip("/")
+        self.timeout = timeout
+        self.a2a_prefix = a2a_prefix
+        self.json_rpc_endpoint = json_rpc_endpoint
+        self._async_client: Optional[AsyncClient] = None
+        self._sync_client: Optional[Client] = None
+
+    # Sync context manager support
```

Comment:
Our AgentOS client doesn't have any context managers. I don't think we need them here for now. Check in our AgentOS client how we use the global client.